File: Python/factorial_using_recurssion.py
def factorial(n):
    return 1 if n == 0 else n * factorial(n - 1)                       # 0! = 1, recursion for n > 0

def main():
    num = int(input("Enter a number: "))
    if num < 0:
        print("Factorial of a negative number doesn't exist.")
    else:
        print(f"Factorial of {num} is {factorial(num)}")

if __name__ == "__main__":
    main()

File: C/Factorial_using_recurssion.c
#include <stdio.h>

// Function to calculate factorial using recursion
unsigned long long factorial(int n) {
    if (n == 0) {                            //0! = 1
        return 1;
    } else {
        return n * factorial(n - 1);         //recursion : n! = n * (n-1)!
    }
}

int main() {
    int num;
    printf("Enter a number: ");
    scanf("%d", &num);

    if (num < 0) {
        printf("Factorial of a negative number doesn't exist.\n");
    } else {
        printf("Factorial of %d is %llu\n", num, factorial(num));
    }

    return 0;
}